### ISSUE TITLE
HTTP OPTIONs does not seems to be working 

### ISSUE THREAD

**Issue by abhinav-jain09** [Azure/Azure-Functions#1020]
Hi Team,

I wrote the HTTP triggered Azure function with spring boot cloud adaptor . I define the http method as POST and Options on the handler class as follows 
```
public class BookHandler extends AzureSpringBootRequestHandler<String, ExecutionResult> {
	@FunctionName("queryHippoContent")
	public ExecutionResult run(
			@HttpTrigger(name = "req", methods = { 	HttpMethod.OPTIONS,
					HttpMethod.POST }, authLevel = AuthorizationLevel.ANONYMOUS)String bar ,
			ExecutionContext context) {

	


	   	return	handleRequest(bar, context);
	}
```

When I invoke the url to run this function with OPTIONS set in the postman tool it retun me null pointer exception with http 500 . Bascailly with the options method this is expecting the value for "bar " varible but I am not passing anything from the Postman becuase its a OPTION call no the reall post call. 

Please help me

**Comment by jeffhollan** [comment]
Is it possible to add an “if bar == null” somewhere in code path for option? I’m not sure if spring boot framework throwing issue or the handler you have on other side of this method call

**Comment by abhinav-jain09** [comment]
@jeffhollan  If I add if bar==null then we are making work around for this problem . If I use spring boot alone without handler its works out of the box meaning if I send request with OPTIONS without any query paramater call goes throw and respose comes as 200 ok. I think we need to find solution how it will work with azure as well . What do you say?

**Comment by jeffhollan** [comment]
I believe if you don't explicitly allow OPTIONS in the attribute the default runtime will do the behavior you want. By specifying you send options into the request and would have to handle null.  Does it work if you remove `HttpMethods.OPTIONS`?  I think it should

**Comment by abhinav-jain09** [comment]
@jeffhollan I did without specifying ``` HttpMethods.OPTIONS``` then I get 404 response . it does not find the resource :-(.

In Ideal case you are right if I do not specify ```HttpMethods.OPTIONS``` it should work but its not working

**Comment by jeffhollan** [comment]
Ok may need to move to the host runtime repo to track the ask to have non-declared OPTIONS return a valid response.  @fabiocav am I crazy for thinking that was existing behavior?

**Comment by MHHenriksen** [comment]
In case it's helpful, for our app we've chosen to create a catch-all function that answers OPTIONS for any path, since we couldn't get CORS with OPTIONS working any other way for all our actual functions.

**Comment by abhinav-jain09** [comment]
@MHHenriksen I am sorry but I do not understand you. Do you mean current functions implementation does not support OPTIONS or the scenario which I mention does not work?

**Comment by abhinav-jain09** [comment]
@MHHenriksen  In ideal case without mapping OPTIONS by default function should work and reply as expected reply of OPTIONS

**Comment by MHHenriksen** [comment]
I'm just speaking from our experience from using functions, I'm not a contributor and don't know everything that does, doesn't, should or shouldn't work :P 

But I do know that simply enabling CORS in the configuration doesn't make your endpoints answer OPTIONS requests, it only adds the usual headers to normal responses. So either you have to manually implement support for OPTIONS on each of your functions (possibly without the normal non-path parameters for the function), or do like we did and create an extra function that binds to a catch-all wildcard path for answering OPTIONS requests for the whole app.

In other words, it works, but not automatically. I slightly agree that the framework should have built-in handling for this if CORS is configured.

**Comment by jeffhollan** [comment]
Let me try to repro this. I *believe* I've done this before without explicitly taking options but am not sure anymore

**Comment by fabiocav** [comment]
@jeffhollan any follow up required on this?

**Comment by jeffhollan** [comment]
I was able to repro this behavior and I think I have the answer.  Apologies for the loooong delay.

The right solution here assuming you don't want to do CORS yourself in code, enable the app service / function CORS settings.  However the request will only be routed automatically back with the right pre-flight stuff if you include the correct headers.  So if I just make an OPTIONS request I get a 404 as well, but if I include origin, host, Access-Control-Request-Method, Access-Control-Request-Headers headers then I get a 200 with the right pre-flight material.

Thanks for reminding me @fabiocav
